lib/streamlit/delta_generator.py

```diff
@@ -903,3 +923,9 @@ def _enqueue_message(msg: ForwardMsg_pb2.ForwardMsg) -> None:
         msg.delta.fragment_id = ctx.current_fragment_id
 
     ctx.enqueue(msg)
+
+
+def _writes_directly_to_sidebar(dg: DG) -> bool:
+    in_sidebar = sidebar_dg in set(dg._anscestors)
+    has_container = bool(len(list(dg._anscestor_block_types)))
```

Comment:
Wouldn't this always be true if we are in the sidebar? Isn't a sidebar a block type?

Reply:
The sidebar is a container, but it's not a block type and is treated more closely to the main dg throughout the code.